### ISSUE TITLE
[Feature]: Implement screen-accurate pack/wand idle fadeout

### ISSUE THREAD

**Issue by Toy203** [gpstar81/GPStar-proton-pack#347]
### What would you like to add in terms of software changes?

It would be cool to have an option and set the proton stream sound volume separately from the rest. My main thing would be to be able to suprise folks with the firing sound without being deafened by the regular pack droning. Just a random thing I know, I let some kids fire the gun and it's fun but I could imagine hearing an unexpected thunderous sound would be even funnier.  Maybe en eeprom settings to keep it just a few notches over the rest of the sounds. 

### Would this request involve any specific hardware?

Nope

### Homework Completed

- [X] I did my part!

**Comment by nomakewan** [comment]
At least for the moment it appears we're out of room for a dedicated setting, but I might try to see if there's a way to "toggle" the system volume to change between volume settings without actually moving what level/option you are on. Something like if you click Intensify it'll change to the next volume setting, while holding it down lets you adjust that setting.

Not going to give any guarantees for this one, but I'll see what can be cooked up for v6.

**Comment by The-Y33TER** [comment]
Somewhat similar, why is the boson dart so quiet? All the other modes and alt fires are a decent volume, but the boson dart is barely audible.

**Comment by nomakewan** [comment]
I just checked the Boson Dart audio files and they're at maximum volume without clipping, exactly as they came from the 2009 video game. It also didn't sound all that different from the Shock Blast sound. Perhaps because only one part of the Boson Dart sound is actually loud, while the rest trails off, unlike other alt fire modes?

Feel free to inspect the audio file for yourself in your audio editor of choice. If you have Proton Stream Impact Effects enabled, the file in use is `411_boson_dart_fire_impact.wav`. If you have this setting disabled, the file in use is `373_boson_dart_fire.wav`.

Likewise, there is no volume reduction in the code for these files. You can confirm this for yourself here by seeing that the volume variable in use is the normal `i_volume_effects` used for all wand sound effects universally: https://github.com/gpstar81/GPStar-proton-pack/blob/main/source/NeutronaWand/NeutronaWand.ino#L2865

**Comment by nomakewan** [comment]
Wanted to update on this and collect thoughts. The issue with implementing this as-asked is really that we're already playing everything at full volume. So to do this as-asked we would actually have to run the entire system at a lower volume than the firing sounds in order for the firing sounds to stand out, which isn't any fun (it would reduce the 'wow' factor of switching someone on).

However, I believe a good compromise is in fact already something we planned on doing for a future release, which is the "screen accurate idle" mode. Basically instead of the idle sound playing on an infinite loop as it is now, you'll get the full volume pack startup, then the idle for a little bit, and then the idle fades out (while the pack lights continue running). This way, when you fire the only sound you'll hear is in fact the firing sound, and it'll come from basically nowhere.

This is how the packs are depicted on-screen because moviegoers would generally be annoyed if every single scene where the packs were on the actors' voices were drowned out by that idle noise. It's something a lot of cosplayers have requested, and I'm certainly happy to implement. Of course it would be selectable by the user in case they prefer their pack always make noise.

Would that work to solve your feature request?

**Comment by Toy203** [comment]
I'd say it'll be even better for usability, having the droning fade out after a bit would help during event tremendously. Maybe user could have input on how much time should pass from switch on to fading out too, maybe folks like the constant humming.

**Comment by nomakewan** [comment]
@Toy203 As of the latest commit in `6.1.Dev`, this function has been implemented. Activate it by going to the Pack Settings in the web UI, and toggle on "Fade Out Idle Loop Sounds". This will cause the pack idle sounds to fade out after 30 seconds.

The idle sounds will return on reboot, or when switching stream modes.

**Comment by Toy203** [comment]
Would it also return momentarily on firing by any chance? (sorry for reopening)
That would be awesome.

**Comment by nomakewan** [comment]
It does not return on _firing_ as in every scene in the film where firing happens, it does not (think the Sedgewick, the rooftop, Afterlife's "Switch me on" test scene, etc).

**Comment by Toy203** [comment]
I can open a new optional feature request for further down the line then? Don't want to be a burden.

**Comment by Toy203** [comment]
Sorry i realize i might be completely off as it should already do what i like. It'll still play firing sound at full volume on firing as it is, it just won't "drone" back for a bit after firing, correct?

**Comment by nomakewan** [comment]
Correct. The firing sounds are completely untouched, and the firing tails still play as expected. The idle drone simply doesn't come back after firing ends.

Similar to how the "Switch me on" scene in Afterlife worked, where the idle faded out just before firing started, then the firing tail had the little whirr down in it.

**Comment by Toy203** [comment]
Awesome! Let's close this down for good.

**Comment by nomakewan** [comment]
Please do feel free to test `6.1.Dev` out on your equipment and see what you think.

Note that the current build in `6.1.Dev` will reset your EEPROMs when flashed (this is by design), and that it requires new audio files for your microSD cards. You can either reload _all_ sounds if that's easier for you, or if you want to piecemeal the updated sounds for `6.1.Dev` are in the [sounds folder](https://github.com/gpstar81/GPStar-proton-pack/tree/v6.1.Dev/sounds/pack-wand) and are the sounds numbered `334`, `352`, `362`, `363`, `441`, and `442`.
